doc/classes/OS.xml

```diff
@@ -760,6 +760,9 @@
 		<constant name="RENDERING_DRIVER_D3D12" value="2" enum="RenderingDriver">
 			The Direct3D 12 rendering driver.
 		</constant>
+		<constant name="RENDERING_DRIVER_METAL" value="3" enum="RenderingDriver">
+			The Metal rendering driver.
+		</constant>
```

Comment:
I added this to be consistent with `RENDERING_DRIVER_D3D12`, but I don't see how this is used anywhere.

Reply:
These don't seem used anywhere indeed, we could flag this enum as deprecated.

It used to be used in 3.x for `OS::get_current_video_driver()` (7a79eee5234), and was refactored for 4.x in f82deaa5b38db124ef8b2f395b54602e3ff8e4ce, but the API that used it no longer exists. I'm not sure what replaced it in 4.x, CC @Calinou.

We also have this piece of dead code to remove eventually:
```
editor/plugins/node_3d_editor_plugin.cpp
5339:   //if (OS::get_singleton()->get_current_video_driver() == OS::RENDERING_DRIVER_OPENGL3) {
```

Reply:
> I'm not sure what replaced it in 4.x, CC @Calinou.

We seem to have:
```
core/os/os.h:
        String get_current_rendering_driver_name() const { return _current_rendering_driver_name; }
        String get_current_rendering_method() const { return _current_rendering_method; }
```
but not exposed to scripting.

Reply:
I have a PR to expose these: https://github.com/godotengine/godot/pull/85430